apps/server/src/modules/task/entity/task.entity.ts

```diff
@@ -1,14 +1,16 @@
-import { Entity, ManyToOne, Property } from '@mikro-orm/core';
-import { ObjectId } from '@mikro-orm/mongodb';
+import { Collection, Entity, ManyToOne, OneToMany, Property } from '@mikro-orm/core';
 import { BaseEntityWithTimestamps, EntityId } from '@shared/domain';
+import { Course } from '@src/entities';
+import type { Submission } from './submission.entity';
 import { LessonTaskInfo } from './lesson-task-info.entity';
 
 interface ITaskProperties {
 	name: string;
 	dueDate?: Date;
 	private?: boolean;
-	parentId: ObjectId;
+	parent?: Course;
 	lesson?: LessonTaskInfo;
+	submissions?: Submission[];
```

Comment:
why not using a collection for submissions?

Reply:
i thought that was better to use basic arrays in the props interface. makes assigning of collections in the constructor easier. otherwise we would have to provide something like `new Collection<Submission>(task, submissions)`.